usaspending_api/common/cache_decorator.py

```diff
@@ -1,13 +1,31 @@
 # -*- coding: utf-8 -*-
 import logging
 
+from django.conf import settings
 from django.db.models import QuerySet
 from rest_framework_extensions.cache.decorators import CacheResponse
+from typing import Any
 from usaspending_api.common.experimental_api_flags import is_experimental_elasticsearch_api
 
 logger = logging.getLogger("console")
 
 
+def contains_queryset(data: Any) -> bool:
+    """Boring non-generator to traverse a complex object and flag any Querysets"""
+    if isinstance(data, QuerySet):
+        return True
+    elif isinstance(data, dict):
+        for v in data.values():
+            if contains_queryset(v):
+                return True
+    elif isinstance(data, list):
```

Comment:
tuples?  sets?  other iterables?

Reply:
Good catch. At first the correction caused problems by significantly increasing recursion depth and hitting the limit. Then I realized strings are Iterables and were blowing up the function calls. It's so obvious in hindsight

Reply:
lol... stupid strings... sorry for causing a headache